int16 mode
int16 pos1
int16 pos2
int16 pos3